pkg/registry/core/adapters/nse_context_client_test.go

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2020 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package adapters_test
+
+import (
+	"context"
+
+	"github.com/networkservicemesh/sdk/pkg/registry/utils/checks/checkcontext"
+
+	"github.com/networkservicemesh/sdk/pkg/registry/core/adapters"
+	"github.com/networkservicemesh/sdk/pkg/registry/core/next"
+	"github.com/networkservicemesh/sdk/pkg/registry/core/streamcontext"
+
+	"testing"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/networkservicemesh/api/pkg/api/registry"
+)
+
+type writeNSEServer struct {
+}
+
+func (w *writeNSEServer) Register(ctx context.Context, service *registry.NetworkServiceEndpoint) (*registry.NetworkServiceEndpoint, error) {
+	ctx = context.WithValue(ctx, testKey, true)
+	return next.NetworkServiceEndpointRegistryServer(ctx).Register(ctx, service)
+}
+
+func (w *writeNSEServer) Find(query *registry.NetworkServiceEndpointQuery, server registry.NetworkServiceEndpointRegistry_FindServer) error {
+	ctx := context.WithValue(server.Context(), testKey, true)
+	return next.NetworkServiceEndpointRegistryServer(server.Context()).Find(query, streamcontext.NetworkServiceEndpointRegistryFindServer(ctx, server))
+}
+
+func (w *writeNSEServer) Unregister(ctx context.Context, service *registry.NetworkServiceEndpoint) (*empty.Empty, error) {
+	ctx = context.WithValue(ctx, testKey, true)
+	return next.NetworkServiceEndpointRegistryServer(ctx).Unregister(ctx, service)
+}
+
+func TestNSEClientPassingContext(t *testing.T) {
+	n := next.NewNetworkServiceEndpointRegistryClient(adapters.NetworkServiceEndpointServerToClient(&writeNSEServer{}), checkcontext.NewNSEClient(t, func(t *testing.T, ctx context.Context) {
+		if _, ok := ctx.Value(testKey).(bool); !ok {
+			t.Error("Context ignored")
+		}
+	}))
+	_, _ = n.Register(context.Background(), nil)
+	_, _ = n.Find(context.Background(), nil)
+	_, _ = n.Unregister(context.Background(), nil)
```

Comment:
Can we add check that errors here are nil?

Reply:
ok, fixed